src/page/TimerPage/hooks/useTimerPageState.ts

```diff
@@ -0,0 +1,291 @@
+
+  /**
+   * 라운드 이동/초기 진입 시 타이머 상태 초기화 및 셋업
+   */
+  useEffect(() => {
+    if (!data) return;
+    const currentBox = data.table[index];
+    timer1.clearTimer();
+    timer2.clearTimer();
+    normalTimer.clearTimer();
+    normalTimer.handleCloseAdditionalTimer();
+
+    // 일반 타이머(중립 발언 등)
+    if (currentBox.boxType === 'NORMAL') {
+      const defaultTime = currentBox.time ?? 0;
+      normalTimer.setDefaultTimer(defaultTime);
+      normalTimer.setTimer(defaultTime);
+    }
+    // 진영별 타이머(찬/반)
+    else if (currentBox.boxType === 'TIME_BASED') {
+      normalTimer.clearTimer();
+      const defaultTotalTimer = currentBox.timePerTeam;
+      const defaultSpeakingTimer = currentBox.timePerSpeaking;
+      [timer1, timer2].forEach((timer) => {
+        timer.setDefaultTime({ defaultTotalTimer, defaultSpeakingTimer });
+        timer.setTimers(defaultTotalTimer, defaultSpeakingTimer);
+        timer.setIsSpeakingTimer(true);
+        timer.setIsDone(false);
+      });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [
+    data,
+    index,
+    timer1.setDefaultTime,
+    timer1.setTimers,
+    timer2.setDefaultTime,
+    timer2.setTimers,
+    normalTimer.setDefaultTimer,
+    normalTimer.setTimer,
+  ]);
+
+  /**
+   * 진영 전환 시, 상대 타이머를 발언 구간에 맞게 초기화
+   */
+  useEffect(() => {
+    if (prosConsSelected === 'cons') {
+      if (timer1.speakingTimer === null) return;
+      timer1.resetTimerForNextPhase();
+    } else if (prosConsSelected === 'pros') {
+      if (timer2.speakingTimer === null) return;
+      timer2.resetTimerForNextPhase();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [prosConsSelected]);
+
+  /**
+   * 각 타이머가 종료 시 자동으로 타이머 일시정지
+   */
+  useEffect(() => {
+    if (timer1.speakingTimer === 0 || timer1.totalTimer === 0) {
+      timer1.pauseTimer();
+    } else if (timer2.speakingTimer === 0 || timer2.totalTimer === 0) {
+      timer2.pauseTimer();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [
+    timer1.speakingTimer,
+    timer1.totalTimer,
+    timer2.speakingTimer,
+    timer2.totalTimer,
+  ]);
+
+  /**
+   * 각 진영의 타이머가 완전히 끝난 경우(isDone 처리)
+   */
+  useEffect(() => {
+    if (prosConsSelected === 'pros') {
+      if (timer1.speakingTimer === null) {
+        if (timer1.totalTimer === 0) {
+          timer1.setIsDone(true);
+        }
+      } else {
+        if (timer1.speakingTimer === 0) {
+          timer1.setIsDone(true);
+        }
+      }
+    } else if (prosConsSelected === 'cons') {
+      if (timer2.speakingTimer === null) {
+        if (timer2.totalTimer === 0) {
+          timer2.setIsDone(true);
+        }
+      } else {
+        if (timer2.speakingTimer === 0) {
+          timer2.setIsDone(true);
+        }
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [
+    prosConsSelected,
+    timer1.totalTimer,
+    timer1.speakingTimer,
+    timer2.totalTimer,
+    timer2.speakingTimer,
+  ]);
```

Comment:
```suggestion
useEffect(() => {
  const selectedTimer = prosConsSelected === 'pros' ? timer1 : timer2;

  const isDone =
    selectedTimer.speakingTimer === null
      ? selectedTimer.totalTimer === 0
      : selectedTimer.speakingTimer === 0;

  if (isDone) {
    selectedTimer.setIsDone(true);
  }
}, [
  prosConsSelected,
  timer1.totalTimer,
  timer1.speakingTimer,
  timer2.totalTimer,
  timer2.speakingTimer,
]);
```
여기도 이렇게 if문을 정리하는 것은 어떠신가요 ?!